materialize-rockset/driver.go

```diff
@@ -35,40 +37,100 @@ func (c *config) Validate() error {
+type collectionSettings struct {
+	RetentionSecs *int64           `json:"retention_secs,omitempty" jsonschema:"title=Retention Period,description=Number of seconds after which data is purged based on event time"`
+	EventTimeInfo *eventTimeInfo   `json:"event_time_info,omitempty" jsonschema:"title=Event Time Info"`
+	ClusteringKey []fieldPartition `json:"clustering_key,omitempty" jsonschema:"title=Clustering Key,description=List of clustering fields"`
+}
 
+func (s *collectionSettings) Validate() error {
+	if s.RetentionSecs != nil && *s.RetentionSecs < 0 {
+		return fmt.Errorf("Retention Period cannot be negative")
+	}
+	if s.EventTimeInfo != nil {
+		return s.EventTimeInfo.Validate()
+	}
+	// nothing to validate on ClusteringKey
 	return nil
 }
 
 type resource struct {
-	Workspace  string `json:"workspace,omitempty"`
-	Collection string `json:"collection,omitempty"`
-	// Configures the rockset collection to bulk load an initial data set from an S3 bucket, before transitioning to
-	// using the write API for ongoing data.
-	// If a previous version of this materialization wrote files into S3 in order to more quickly backfill historical
-	// data, then this value should contain configuration about the integration.  See: https://go.estuary.dev/rock-bulk
-	// If a bulk loading integration is not being used, then this should be undefined.
-	InitializeFromS3 *cloudStorageIntegration `json:"initializeFromS3,omitempty"`
-	MaxBatchSize     int                      `json:"maxBatchSize,omitempty"`
+	Workspace string `json:"workspace,omitempty" jsonschema:"title=Workspace,description=The name of the Rockset workspace (will be created if it does not exist)"`
+	// The name of the Rockset collection (will be created if it does not exist)
+	Collection string `json:"collection,omitempty" jsonschema:"title=Rockset Collection,description=The name of the Rockset collection (will be created if it does not exist)"`
+	// Configures the rockset collection to bulk load an initial data set from an S3 bucket, before
+	// transitioning to using the write API for ongoing data. If a previous version of this
+	// materialization wrote files into S3 in order to more quickly backfill historical data, then
+	// this value should contain configuration about the integration.  See:
+	// https://go.estuary.dev/rock-bulk If a bulk loading integration is not being used, then this
+	// should be undefined.
+	InitializeFromS3 *cloudStorageIntegration `json:"initializeFromS3,omitempty" jsonschema:"title=Backfill from S3" jsonschema_extras:"advanced=true"`
+
+	AdvancedCollectionSettings *collectionSettings `json:"advancedCollectionSettings,omitempty" jsonschema:"title=Advanced Collection Settings" jsonschema_extras:"advanced=true"`
+	// TODO: add advanced annotation to collectionSettings and InitializeFromS3
+	// TODO: set insert_only when creating collections, based on presence of delta_updates
```

Comment:
That TODO was actually pretty nonsensical. Just because `delta_updates` is `true`, it doesn't mean that we won't still see multiple documents with the same key. I did just push a commit that adds `insert_only` as an advanced collection creation option, though, so users can set it if they need.